tests/trainer/test_trainer.py

```diff
@@ -666,71 +667,29 @@ def test_benchmark_option(benchmark_, deterministic, expected):
 
 
 @pytest.mark.parametrize("ckpt_path", (None, "last"))
-@pytest.mark.parametrize("fn", ("fit", "validate"))
-@mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": "1"})
-def test_checkpoint_path_input_last_fault_tolerant(tmpdir, ckpt_path, fn):
-    should_signal = True
-
-    class ExitGracefullyException(Exception):
-        pass
-
-    class TestModel(BoringModel):
-        def validation_step(self, batch, batch_idx):
-            self.log("foo", -batch_idx)
-            if should_signal and batch_idx == 1:
-                raise ExitGracefullyException
-            return super().validation_step(batch, batch_idx)
-
-        def training_step(self, batch, batch_idx):
-            if should_signal and batch_idx == 1:
-                raise ExitGracefullyException
-            return super().training_step(batch, batch_idx)
-
-    model = TestModel()
-    model.test_epoch_end = None
-    mc = ModelCheckpoint(monitor="foo")
-    trainer = Trainer(
-        max_epochs=2,
-        limit_val_batches=3,
-        enable_progress_bar=False,
-        default_root_dir=tmpdir,
-        callbacks=[mc],
-    )
-    assert trainer.ckpt_path is None
-    trainer_fn = getattr(trainer, fn)
-
-    from pytorch_lightning.callbacks.fault_tolerance import _FaultToleranceCheckpoint
-
-    ft_checkpoints = [cb for cb in trainer.callbacks if isinstance(cb, _FaultToleranceCheckpoint)]
-    ft_ckpt_path = ft_checkpoints[0].ckpt_path
+@pytest.mark.parametrize("fn", (TrainerFn.FITTING, TrainerFn.VALIDATING))
+def test_checkpoint_path_input_last_fault_tolerant(ckpt_path, fn):
```

Comment:
This rewritten test now fails. The reason for that is that loading of last checkpoint checks whether the file exists and the fault tolerant checkpoint does not exist

Reply:
I fixed it with a simple addition